kvstore/pebble/kvstore.go

```diff
@@ -0,0 +1,163 @@
+package pebble
+
+import (
+	"errors"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/vfs"
+)
+
+var _ PebbleKVStore = &pebbleKVStore{}
+
+type pebbleKVStore struct {
+	db *pebble.DB
+}
+
+// NewKVStore creates a new PebbleKVStore instance.
+// If path is empty, it creates an in-memory store.
+func NewKVStore(path string) (PebbleKVStore, error) {
+	store := &pebbleKVStore{}
+
+	opts := &pebble.Options{}
+	if path == "" {
+		opts.FS = vfs.NewMem()
+	}
```

Comment:
Same as with the `badger` comment above. Add a comment saying to enable database configurations to be passed in during creation.

Reply:
Great suggestion, let me do that :)